File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.py
def solution(n):
    d = [0] * (n + 1)
    d[1], d[2], d[3] = 1, 2, 3
    
    if n <= 3: return d[n]
    
    for i in range(4, n + 1):
        d[i] = (d[i - 2] + d[i - 1]) % 1000000007
    
    return d[n]

File: 프로그래머스/2/12900. 2 x n 타일링/2 x n 타일링.js
function solution(n) {
    const D = 1000000007;
    const table = [0, 1, 2];
    
    if (n < 3) return table[n];
    
    for (let i = 3; i <= n; i++) {
        table.push((table[i - 1] % D) + (table[i - 2] % D))
    }

    return table[n] % D;
}
